src/writer.rs

```diff
@@ -0,0 +1,58 @@
+use std::io::{Write, Error, ErrorKind};
+use std::io;
+use InflateStream;
+
+pub struct InflateWriter<W: Write> {
+    inflater: InflateStream,
+    writer: W
+}
+
+impl<W: Write> InflateWriter<W> {
+    pub fn new(w: W) -> InflateWriter<W> {
+        InflateWriter { inflater: InflateStream::new(), writer: w }
+    }
+
+    pub fn finish(mut self) -> io::Result<W> {
```

Comment:
I think it would probably be a good idea to use the same naming as [flate2](https://docs.rs/flate2/0.2.17/flate2/write/struct.DeflateEncoder.html).